app/logic/Calendar/JSON/JSONEvent.ts

```diff
@@ -0,0 +1,107 @@
+import { Event } from "../Event";
+import { RecurrenceRule } from "../RecurrenceRule";
+import { PersonUID, findOrCreatePersonUID } from "../../Abstract/PersonUID";
+import { appGlobal } from "../../app";
+import { sanitize } from "../../../../lib/util/sanitizeDatatypes";
+import { assert } from "../../util/util";
+import { Collection } from "svelte-collections";
+
+export class JSONEvent {
+  static save(event: Event): any {
+    assert(event.calendar?.id, "Need calendar ID to save the event");
+    let json: any = {};
+    json.pID = event.pID;
```

Comment:
What do we use for id here? `Event.id` is readonly so we can't read and set it when reading. But we also can't use pID because then local calendar would not work

Reply:
You're right. Remove the `readonly` flag in `Event` class.